CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticViewTopComponent.java

```diff
@@ -74,6 +75,7 @@
     private final AnalyticViewPane analyticViewPane;
     private final AnalyticController analyticController;
     private boolean suppressed = false;
+    private Graph currentGraph = null;
```

Comment:
There is potential for a memory leak to appear if the `currentGraph` is not set to `null` when the graph closes. I am speculating but I will do some testing to confirm @aldebaran30701.

Reply:
How does it sound if I store the current graph's ID instead?

Reply:
That sounds like a better idea though we haven't depended on the graph ID for much so I am not sure if it is the best approach. If it works then that does sound like a really good way forward and perhaps we can adopt it across more of the framework.

I remember seeing the graph ID change after plugins run (but I need to double check my memory). If that's the case (which I don't think so) would be a problem.

Reply:
Hi @cygnus-x-1,  Would you be able to confirm for us if the graph id changes between plugin runs? Or subsequently offer a different way of tracking the current graph without memory leaks.

Reply:
I don't know whether or not it helps avoid memory leaks but having played with the rest api previously, I believe the graph Id remains the same between plugin runs (even between Constellation loads)

Reply:
>...(even between Constellation loads)

Not sure on this bit @antares1470. The graph UUID is not stored in the star file so I believe a random one is issued on load.

Reply:
I believe a graph id should be reliable as long as that graph is in memory.

Reply:
Right. Like I said, it's been a while so I was only going off the top of my head

Reply:
> If the graph is closed and opened again, it'll have a new id

I'd think that would cross over into a future state implementation. This (keeping the id) would work for the sake of not clearing the results table.